ReactNativeClient/lib/services/UndoRedoService.ts

```diff
@@ -0,0 +1,48 @@
+export class UndoRedoService {
+	totalKeyEventCount:any;
+	totalClickEventCount:any;
+	textHistory:string[];
+	text:string;
+
+	constructor() {
+		this.totalKeyEventCount = 0;
+		this.totalClickEventCount = 0;
+		this.textHistory = [];
+		this.text = '';
+	}
+
+	// It will, increment totalKeyEventCount on text change and get the text
+	// to store with refrence to totalKeyEventCount inside textHistory[]
+	handleText(text:string) {
+		this.totalKeyEventCount++;
+		this.textHistory[this.totalKeyEventCount] = text;
+	}
```

Comment:
This class is hard to make sense of and it's good you've added comments, however comments should explain **why** the code does something, **not what it does**.

Literally all your comments repeat what the code does. I can read code, I know ++ means "increment", and ">" means "greater than", but I don't know why your class needs to keep track of event count, why array elements are added directly by index, and not with "push", why elements are not discarded after undo, why you need to store a "text" value when the undo stack should give already give you that value, etc.

totalKeyEventCount and totalClickEventCount are of "any" type, but they must be numbers based on your code.

I'm sorry I'm not very keen on the design of this class.